probables/cuckoo/cuckoo.py

```diff
@@ -97,6 +98,62 @@ def __init__(
             msg = "CuckooFilter: failed to load provided file"
             raise InitializationError(msg)
 
+        self._error_rate = float(self._calc_error_rate())
+
+    @classmethod
+    def init_error_rate(
+        cls,
+        error_rate: float,
+        capacity: int = 10000,
+        bucket_size: int = 4,
+        max_swaps: int = 500,
+        expansion_rate: int = 2,
+        auto_expand: bool = True,
+        hash_function: typing.Optional[SimpleHashT] = None,
+    ):
+        """Initialize a simple Cuckoo Filter based on error rate
+
+        Args:
+            error_rate (float):
+            capacity (int): The number of bins
+            bucket_size (int): The number of buckets per bin
+            max_swaps (int): The number of cuckoo swaps before stopping
+            expansion_rate (int): The rate at which to expand
+            auto_expand (bool): If the filter should automatically expand
+            hash_function (function): Hashing strategy function to use \
+            `hf(key)`
+        Returns:
+            CuckooFilter: A Cuckoo Filter object"""
+        cku = CuckooFilter(
+            capacity=capacity,
+            bucket_size=bucket_size,
+            auto_expand=auto_expand,
+            max_swaps=max_swaps,
+            expansion_rate=expansion_rate,
+            hash_function=hash_function,
+        )
+        # reset fingerprint based on error rate
+        cku._fingerprint_size = int(math.ceil(math.log(1.0 / error_rate, 2) + math.log(2 * cku.bucket_size, 2)))
+        cku._error_rate = error_rate
+        return cku
+
+    @classmethod
+    def load_error_rate(cls, error_rate: float, filepath: str, hash_function: typing.Optional[SimpleHashT] = None):
+        """Initialize a previously exported Cuckoo Filter based on error rate
+
+        Args:
+            error_rate (float):
```

Comment:
Are the types really needed here?

Reply:
I would like to keep them for now until I can find a good way to have Shpinx populate it when building the documentation.